port/win/io_win.cc

```diff
@@ -680,12 +680,9 @@ inline IOStatus WinRandomAccessImpl::ReadImpl(uint64_t offset, size_t n,
                                               char* scratch) const {
   // Check buffer alignment
   if (file_base_->use_direct_io()) {
-    if (!IsSectorAligned(static_cast<size_t>(offset)) ||
-        !IsAligned(alignment_, scratch)) {
-      return IOStatus::InvalidArgument(
-          "WinRandomAccessImpl::ReadImpl: offset or scratch is not properly "
-          "aligned");
-    }
+    assert(IsSectorAligned(file_base_->GetSectorSize(),
+                           static_cast<size_t>(offset)));
+    assert(IsAligned(alignment_, scratch));
```

Comment:
Why are the asserts here better than the old method of returning a bad status?  The assert will not be triggered in non-debug mode, whereas the old code would return bad status regardless of the compilation mode.

Reply:
similarly like io_poix.cc, using assert provides a slight better performance under retail build. If the size is not aligned, it fails call the system api anyway, so adding an extra checking is not really necessary.